libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -114,7 +113,18 @@ const getSharepointData = async (url) => {
   };
 };
 
-const getProjectExcelPath = (url) => url.pathname.replace('.json', '.xlsx');
+const getFilePath = async ({ ref, repo, owner, excelRef } = {}) => {
+  const status = await fetch(`${ADMIN}/status/${owner}/${repo}/${ref}?editUrl=${excelRef}`);
+  const statusJson = await status.json();
+  return (new URL(statusJson?.preview?.url)).pathname;
+};
+
+const getPreviewUrl = ({ owner, repo, ref, filePath } = {}) => fetch(
+  `${ADMIN}/preview/${owner}/${repo}/${ref}${filePath}`,
+  { method: 'POST' },
+)
+  .then((res) => res.json())
+  .then((res) => new URL(res.preview.url));
```

Comment:
we should catch potential errors

Reply:
True, we might have unexpected errors that we'll want to catch. Made the change 👍